client/prometheus.go

```diff
@@ -51,6 +51,31 @@ func (c *Prometheus) Close() error {
 	return nil
 }
 
+// buildIncrement calls incrementer factory if incrementer was not created before
+func (c *Prometheus) buildIncrement(metric, metricTotal string) {
```

Comment:
I'd do `getIncrement(name string)` instead to avoid accessing the map from other parts of the code, so it would be somethign like:

```golang
func (c *Prometheus) getIncrement(name string) incrementer.Incrementer {
    c.Lock()
    defer c.Unlock()

    increment, ok := c.increments[name]
    if !ok {
        increment = c.incFactory.Create()
       c.increments[metric] = increment
    }

    return increment
}
```